### PR TITLE
chore: Update dfx to version 0.10.1

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
Deploying an sns test environment involves uploading wasm files using dfx.  Large arguments such as this are not supported by dfx 0.9.2.  They will be supported in dfx `0.10.2`.  Updating to `0.10.1` now reduces the risk of delay updating to `0.10.2` after it has been released.

# Changes
* Update the dfx version to 0.10.1
* Specified the subnet type in `dfx.json`, as message sizes are now checked by `dfx run` and the default subnet type is `application` which is too small.

# Tests
- [x] CI works
- [x] Local deployments to dfx work
- [x] Deployments to a testnet work (deployed to small11)

### CODE REVIEW

File: dfx.json [hunk 1]
```diff
@@ -117,6 +117,9 @@
     }
   },
   "defaults": {
+    "replica": {
+      "subnet_type": "system"
```
Comment (bitdivine) [inline dfx.json:121]: Message types are now enforced.  The default subnet type is application, for which nns canisters and our own nns-dapp are too large, so deployment fails unless we set the correct value.

File: deploy.sh [hunk 2]
```diff
@@ -224,7 +224,7 @@ if [[ "$DEPLOY_NNS_DAPP" == "true" ]]; then
   #        includes other canisters for testing purposes.  If testing you MAY wish
   #        to deploy these other canisters as well, but you probbaly don't.
   dfx canister --network "$DFX_NETWORK" create nns-dapp --no-wallet || echo "canister may have been created already"
-  dfx deploy --network "$DFX_NETWORK" nns-dapp
+  dfx deploy --network "$DFX_NETWORK" nns-dapp --no-wallet
```
Comment (bitdivine) [inline deploy.sh:227]: This now needs to be set explicitly.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## Unreleased
+* Update dfx to version 10.1
```
Comment (bitdivine) [inline CHANGELOG.md:4]: If we fill in the changelog as we go, deployment should be quicker! :-)

File: dfx.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "dfx": "0.9.2",
+  "dfx": "0.10.1",
```
Comment (bitdivine) [inline dfx.json:2]: This is the key change.

**Review by peterpeterparker — APPROVED** [review]

File: dfx.json (hunk 4 above)
Comment (lmuntaner) [reply dfx.json:2]: 🤯

**Review by lmuntaner — APPROVED** [review]

**Pull request merged**